frontend/src/pages/class/[classId]/session/[sessionId]/join.tsx

```diff
@@ -16,70 +17,152 @@ import {
   isStudentLocallyAuthenticated,
 } from "@/contexts/AuthenticationContext";
 import { UpdateAuthenticationContext } from "@/contexts/UpdateAuthenticationContext";
+import { WebSocketContext } from "@/contexts/WebSocketProvider";
+import { StudentAuthenticationRequestContent } from "@/types/websocket-events";
+import { decodeBase64, encodeBase64 } from "@/utilities/crypto/base64";
 import StudentKeyPair from "@/utilities/crypto/StudentKeyPair";
 import styled from "@emotion/styled";
 import { useRouter } from "next/router";
-import { useCallback, useContext, useEffect } from "react";
+import { useCallback, useContext, useEffect, useRef } from "react";
 import { Col, Container } from "react-bootstrap";
 import { FormattedMessage } from "react-intl";
 
 const SubHeader = styled.h2`
   margin-bottom: 1rem;
 `;
 
+const JoinSessionContent = ({
+  classId,
+  sessionId,
+}: {
+  classId: number;
+  sessionId: number;
+}) => {
+  const authenticationContext = useContext(AuthenticationContext);
+  const router = useRouter();
+
+  const {
+    data: session,
+    error: sessionError,
+    isLoading: isLoadingSession,
+  } = useClassSession(classId, sessionId);
+
+  const onJoinSession = useCallback(async () => {
+    if (
+      !classId ||
+      !session ||
+      session.tasks.length <= 0 ||
+      !isStudentFullyAuthenticated(authenticationContext, session.id)
+    ) {
+      return;
+    }
+
+    // once this is done, the student is fully authenticated and can join the session
+    router.replace(
+      `/class/${classId}/session/${session.id}/task/${session.tasks[0].id}/solve`,
```

Comment:
same remark, how do we keep the URIs in sync across different parts of the codebase?
Should we add a tiny library just for this? 🤔

Reply:
what would that library do? I mean we could write a small script ourselves that for each file in pages/ creates a function in a typescript file based on the file path which Next.js uses for routing?

Reply:
https://classroomrt.atlassian.net/browse/CRT-60